core/write.go

```diff
@@ -9,7 +9,15 @@ import (
 	"github.com/goto/entropy/pkg/errors"
 )
 
-func (svc *Service) CreateResource(ctx context.Context, res resource.Resource) (*resource.Resource, error) {
+type Options struct {
+	DryRun bool
+}
+
+func WithDryRun(dryRun bool) Options {
+	return Options{DryRun: dryRun}
+}
+
```

Comment:
We can do

```suggestion
func WithDryRun() Options {
	return Options{DryRun: true}
}

```
and by default is false, so if it is passed, it means enabling it to true

Reply:
yeah that looks better
But cmiiw in this case the handler will have to have a if else statement like this?

```
	if request.GetDryRun() {
		result, err = server.resourceSvc.CreateResource(ctx, *res, core.WithDryRun())
		if err != nil {
			return nil, serverutils.ToRPCError(err)
		}
	} else {
		result, err = server.resourceSvc.CreateResource(ctx, *res)
		if err != nil {
			return nil, serverutils.ToRPCError(err)
		}
	}
```

Reply:
hm right, lets keep it as you proposed then